src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -497,6 +507,7 @@ private void closeTempFile() throws IOException {
     for (String encodedPartition: tempFiles.keySet()) {
       closeTempFile(encodedPartition);
     }
+    lastRotate = System.currentTimeMillis();
```

Comment:
Is this the right time to do this? I'm wondering if we can end up with closing the file, then running into an issue where it takes quite awhile to actually get the file committed (e.g. we happen to hit HDFS issues then). Then I think the following period wouldn't have been processing messages for as long because we already effectively started it here.

I think we should go entirely one way or the other: if we set `lastRotate` before we've finished the rotation, we should do it immediately when we start (resulting behavior: we rotate every `rotateIntervalMs` regardless of how long committing takes), or we should wait to set `lastRotate` until we're absolutely sure we've finished committing (resulting behavior: process messages for `rotateIntervalMs` before rotating, regardless of how long commits take).